pkg/apis/core/v1alpha1/federationconfig_types.go

```diff
@@ -53,6 +53,13 @@ type FeatureGatesConfig struct {
 	Enabled bool   `json:"enabled,omitempty"`
 }
 
+type ClusterHealthCheckConfig struct {
+	PeriodSeconds    metav1.Duration `json:"period-seconds,omitempty"`
```

Comment:
If your intention is to require the use of seconds, I think `int` would be a more appropriate data type.  Is there precedent here that suggests assuming seconds vs allowing any value to be used?  Same comment on `TimeoutSeconds` below.

Reply:
Agree, `int` is more appropriate data type. Have updated now. PTAL